component/src/main/java/org/wso2/carbon/identity/authenticator/office365/Office365Authenticator.java

```diff
@@ -159,8 +159,17 @@ public String getContextIdentifier(HttpServletRequest request) {
      * Check whether the authentication or logout request can be handled by the authenticator
      */
     @Override
-    public boolean canHandle(HttpServletRequest arg0) {
-        return true;
+    public boolean canHandle(HttpServletRequest request) {
+
+        if (request.getParameter(Office365AuthenticatorConstants.CODE) != null &&
```

Comment:
@NShani what is the reason checking on the these parameters?

I have checked the couple of other similar authenticators that has extended to OIDC authenticator. Those ones simply avoid overriding the canHandle() method.

* https://github.com/wso2-extensions/identity-outbound-auth-yahoo/blob/master/components/org.wso2.carbon.identity.application.authenticator.yahoo/src/main/java/org/wso2/carbon/identity/application/authenticator/yahoo/YahooOAuth2Authenticator.java
* https://github.com/wso2-extensions/identity-outbound-auth-google/blob/master/components/org.wso2.carbon.identity.application.authenticator.google/src/main/java/org/wso2/carbon/identity/application/authenticator/google/GoogleOAuth2Authenticator.java

Reply:
Changed as per this comment.